### PR TITLE
player has item condition improvement

### PR DESCRIPTION
**Pull request by blohod**
## Purpose
Fix issue with item location selection (only first slot worked) and add minimum item amount option.

### CODE REVIEW

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Regions/src/main/java/au/com/mineauz/minigamesregions/conditions/PlayerHasItemCondition.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Narimm/Minigames/pullRequest?prid=5499527&bid=18024022)

File: Regions/src/main/java/au/com/mineauz/minigamesregions/conditions/PlayerHasItemCondition.java [hunk 1]
```diff
@@ -29,6 +29,8 @@
     private final StringFlag where = new StringFlag("ANYWHERE", "where");
     private final IntegerFlag slot = new IntegerFlag(0, "slot");
     
+    private final IntegerFlag amount = new IntegerFlag(1, "amount");
+
```
Comment (Narimm) [inline Regions/src/main/java/au/com/mineauz/minigamesregions/conditions/PlayerHasItemCondition.java:33]: Can we clean up this commit to remove all these extra line breaks?

File: Regions/src/main/java/au/com/mineauz/minigamesregions/conditions/PlayerHasItemCondition.java [hunk 2]
```diff
@@ -51,6 +53,8 @@ public void describe(Map<String, Object> out) {
         out.put("Where", where.getFlag());
         out.put("Slot", slot.getFlag());
         
+        out.put("Amount", slot.getFlag());
+
```
Comment (Narimm) [inline Regions/src/main/java/au/com/mineauz/minigamesregions/conditions/PlayerHasItemCondition.java:57]: here as well

File: Regions/src/main/java/au/com/mineauz/minigamesregions/conditions/PlayerHasItemCondition.java [hunk 3]
```diff
@@ -273,27 +289,28 @@ public Material getValue() {
...
             }
-            
+
             @Override
             public String getValue() {
                 return WordUtils.capitalizeFully(where.getFlag());
             }
         }, Arrays.asList("Anywhere", "Hotbar", "Main", "Armor", "Slot")));
         m.addItem(slot.getMenuItem("Slot", Material.DIAMOND, 0, 35));
+
```
Comment (Narimm) [inline Regions/src/main/java/au/com/mineauz/minigamesregions/conditions/PlayerHasItemCondition.java:299]: extra lines??

File: Regions/src/main/java/au/com/mineauz/minigamesregions/conditions/PlayerHasItemCondition.java [hunk 4]
```diff
@@ -132,15 +132,28 @@ private boolean check(MinigamePlayer player) {
...
+                i += 1;
                 continue;
-            if (slot.getType() == material && slot.getAmount() >= minAmount) {
-                
+            }
+            if (slot != null) {
+
+                i += 1;
+
+                if (slot.getType() != material) {
```
Comment (Narimm) [inline Regions/src/main/java/au/com/mineauz/minigamesregions/conditions/PlayerHasItemCondition.java:146]: These should be combined into 1 check

**Comment by addstar** [comment]
Closing this out - Stale PR, had no response/updates in a long time and too many merge conflicts now. Re-submit if you still want to contribute it :slightly_smiling_face:

**Pull request closed without merging**